lib/src/configuration.cc

```diff
@@ -703,54 +757,65 @@ void Configuration::setDefaultValues()
 
 void Configuration::parseConfigFile()
 {
-    if (!lth_file::file_readable(config_file_))
+    if (!lth_file::file_readable(config_file_)) {
+        LOG_DEBUG("Config file {1} absent", config_file_);
         return;
+    }
 
-    lth_jc::JsonContainer config_json;
+    using boost::algorithm::ends_with;
+    if (!ends_with(config_file_, ".conf") && !ends_with(config_file_, ".json")) {
+        // This is required by the HOCON parser.
+        throw Configuration::Error {
+            _("config-file {1} must end with .conf or .json", config_file_) };
+    }
 
+    hocon::shared_config config;
     try {
-        config_json = lth_jc::JsonContainer(lth_file::read(config_file_));
-    } catch (lth_jc::data_parse_error& e) {
+        LOG_TRACE("Parsing {1}", config_file_);
+        config = hocon::config::parse_file_any_syntax(config_file_)->resolve();
+    } catch (hocon::config_exception& e) {
         throw Configuration::Error {
-            lth_loc::translate("cannot parse config file; invalid JSON") };
+            _("cannot parse config file: {1}", e.what()) };
```

Comment:
What is the difference between `_()` and `lth_loc::format()`?

Reply:
I see both of them bing used throughout the file.

Reply:
One's an alias for the other. I can revert to format to be consistent.

Reply:
Ha, not much!
https://github.com/puppetlabs/leatherman/blob/master/locale/inc/leatherman/locale/locale.hpp#L159-L163

Reply:
I don't mind using either but it's less confusing when only one is used consistently.
So I'm +1 for consistency.

Reply:
Updated.